pkg/pillar/types/dns.go

```diff
@@ -683,6 +683,9 @@ func getLocalAddrListImpl(dns DeviceNetworkStatus,
 
 // Check if an interface name is a port owned by nim
 func IsPort(dns DeviceNetworkStatus, ifname string) bool {
+	if ifname == "" {
+		return false
```

Comment:
We cannot make this assumption since we allow cellular modems to have empty interface name in the device model: https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/zedagent/parseconfig.go#L1137-L1162

Reply:
@milan-zededa but do we want empty ifname to match all/any port with an empty ifname in that case?

Reply:
No, the fix for `IsPort` (to handle wwan adapter with undefined interface name) could be to include PCI and USB addresses from the IoBundle as arguments, and match NetworkPortStatus by non-empty addresses.

Reply:
pushed up a change to incorporate @milan-zededa suggestion.

Reply:
Thanks @milan-zededa